pkg/controller/virtinformers.go

```diff
@@ -488,8 +491,16 @@ func (f *kubeInformerFactory) KubeVirtPod() cache.SharedIndexInformer {
 	})
 }
 
-func (f *kubeInformerFactory) KubeVirtNode() cache.SharedIndexInformer {
+func (f *kubeInformerFactory) KubeVirtNode(nodeName string) cache.SharedIndexInformer {
 	return f.getInformer("kubeVirtNodeInformer", func() cache.SharedIndexInformer {
+		fieldSelector := fields.OneTermEqualSelector("metadata.name", nodeName)
+		lw := NewListWatchFromClient(f.clientSet.CoreV1().RESTClient(), "nodes", k8sv1.NamespaceAll, fieldSelector, labels.Everything())
+		return cache.NewSharedIndexInformer(lw, &k8sv1.Node{}, f.defaultResync, cache.Indexers{})
+	})
+}
+
```

Comment:
We've tried to intentionally avoid having virt-handler watch nodes. That's why the node labeller is designed the way it is today.

Nodes are written to a lot. a whole lot. Ideally we don't need to be streaming all that node data from the api server for every update to the node because we really don't need to react to any node changes on that kind of granularity. Also, this means if we had a controller which queued every time a node change occurred, that reconcile loop would be spinning quite often. <--- in the grand scheme of things, none of that is a huge deal, but the point is the design is intentional.

The node labeller is a best effort controller. We set some labels on the node when virt-handler starts up, then we periodically ensure that those labels stay set by enqueing the host key periodically. It's not like we're dealing with dynamic tunings that are changing throughout the lifecycle of the node. That periodic sync exists just to make sure the node labels remain accurate and don't drift over time.

Reply:
I think that the labeller needs to watch KubevirtCR anyway because of `obsoleteCpuModel` field

Reply:
> The fact that node-labeller reacts to kubevirt CR updates is also new (introduced in this PR).

this logic already exists today.
https://github.com/kubevirt/kubevirt/blob/f58c65a5df922054bf9988a0413aca153b56a7c6/pkg/virt-handler/node-labeller/node_labeller.go#L97

> So, to sum up, I will:
> Make node labeller stop watching the nodes
> Keep watching Kubevirt CR
> Bring back the periodic reconciliation
> Does that sound right?

yes. and I'm pretty certain that's exactly how it works today without this PR.

Reply:
@davidvossel @iholder-redhat @xpivarc 
Please correct me if i'm wrong but i think that we need kubevirt informer to actually trigger the callback function on kubevirtCR changes.